### ISSUE TITLE
connection to KeePassXC breaks after Tor Browser updated itself

### ISSUE THREAD

**Issue by o0nd7ots** [keepassxreboot/keepassxc#9679]
## Overview
KeePassXC plugin for Tor Browser worked fine until I decided to update the Tor Brrowser.
I did it using the built-in tool, and the browser was installed from my distro's package manager.

It would not connect.

## Steps to Reproduce
1. Install Tor Browser from any package manager
2. Open it and install the update (12.4 to 12.5)
3. Install the KeePassXC Addon
4. Launch KeePassXC and unlock a database
5. Turn on Browser Integration with Tor Browser
6. Try to connect to database

## Expected Behavior
Extension asks for uniqe name.

## Actual Behavior
It does not connect

## Context
It put the file into `./.local/share/torbrowser/tbb/x86_64/tor-browser_en-US/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts/`, but it did not work.

KeePassXC - 2.7.5
Revision: 2.1? (what does this mean?)

Operating System: Linux
Desktop Env: KDE
Windowing System: Wayland

## Workaround
I symlinked my 
`~/.mozilla/native-messaging-hosts` 
to
`~/.local/opt/tor-browser/app/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts`
(making the .mozilla folder).

**Comment by varjolintu** [comment]
> Go to application settings, select browser tab, go to advanced. There you can define a custom browser, place the new path for Tor.
> 
> @varjolintu might need to do some folder tests for Tor to see which one to plop the config into.

I'll test the path Tor currently uses. It can vary by how it's installed. `~/.local/opt/tor-browser/app/Browser/TorBrowser/Data/Browser/.mozilla/native-messaging-hosts` in the issue message would be pretty optimal place for a default, if that is what is used nowadays.

**Comment by o0nd7ots** [comment]
@droidmonkey I find the workaround that you suggest worse than mine and it is still a workaround. I suggest that other people use mine. Nevertheless, thanks for the comment.

@varjolintu I think that the `.local/opt...` path is optimal. Would it hurt to put the file or symlinks in many places?

**Comment by droidmonkey** [comment]
Umm it's not a workaround it is a solution that keeps everything in app instead of creating symlinks outside the app. How is that worse?

Tor changing their config file location is not our problem, they are wildly inconsistent across platforms and even across versions now.

**Comment by o0nd7ots** [comment]
I agree that creating symlinks is not the preferred solution, and I consider it a workaround as it works around how the KeePassXC works. Now that I have reconsidered I think it might be a better workaround, but if we are working in-app it would be better to create a complete solution.

I think that if users have the option to select Tor Browser as a browser to integrate with and it does not work so they need to specify it as a _custom browser_ the Tor Browser option is not well supported and that is a in-app workaround.
As KeePassXC is an open source project that I care about I decided to 

I agree that Tor is not easy to work with, but I think that there either should be a disclaimer explaining end users why the integration might not work, and why it is not KeePassXC's fault.
Alternatively, what I consider to be a better solution, KeePass could either add the file everywhere or run tests and determine where is the optimal place to put the file. Maybe if it does not work it could put (or move?) the `native-messaging-hosts` to an alternative location.

I know some C and C++. If you will show me where in the source code the process of putting the file happens I could gladly try to develop a permanent solution. I really appreciate your work and I would like to help somehow.

**Comment by varjolintu** [comment]
@o0nd7ots We have been thinking about an alternative approach to the script installing. In optimal situation there could be just "Enable Browser Integration" checkbox which automatically detects all installed and supported browsers, and installs the script files for them. Obviously we haven't done that yet because the problem mostly affects to Tor Browsers. So this kind of feature would be most beneficial to implement just for Tor.

The class that handles the script file installing is: https://github.com/keepassxreboot/keepassxc/blob/develop/src/browser/NativeMessageInstaller.cpp

**Comment by o0nd7ots** [comment]
I think that a unified checkbox would be very handy but I can see why it might not be the top priority.
I will try to produce a usable solution for the Tor Browser, but I can already tell that it will take me some time, as this is the most complicated program that I have worked with yet.
Thanks for the info!
